lib/HLSL/HLOperationLowerExtension.cpp

```diff
@@ -935,7 +961,15 @@ class CustomResourceLowering
             if (ArgInfo.HighLevelArgIndex < CI->getNumArgOperands())
             {
                 Value *Arg = CI->getArgOperand(ArgInfo.HighLevelArgIndex);
-                if (ArgInfo.HasVectorIndex)
+                if (ConvertArgToStruct)
```

Comment:
I think we should check this after checking for the `ArgInfo.HasVectorIndex`. It would be useful to be able to explicitly explode the vector with the custom overload string and then pack into a struct by default.

Should also add a test to include coverage of this case. Should be fairly easy to add another vector argument that is explicitly exploded by the custom lowering string in the test.

Reply:
Thinking more about this I actually don't think we should even have the `ConvertArgToStruct` bool. We should always convert vector to structs if there is no vector index.

Reply:
That changes custom resource lowering as well, though, since it uses the same path

Reply:
Yeah I think that's ok. We should not be generating vectors in this lowering.